tools/backup_analysis.py

```diff
@@ -127,38 +223,115 @@ def compare_file(dir1, dir2, filename, cmp_size, granularity):
     else:
         change = "remained equal"
         change_diff = 0
+    # Print the time even if no changes because we may want to know how long it took to not
+    # see any changes. Only print if the time is not identical.
+    if start != end:
+        print(f'{filename}: time: started {start} completed {end}')
+    if total_bytes_diff == 0:
+        # If the file is unchanged return now.
+        print(f'{filename}: is unchanged')
+        return
+
+    # Otherwise print out the change information.
     if change_diff != 0:
         print(f'{filename}: size: {f1_size} {f2_size} {change} by {change_diff} bytes')
     else:
         print(f'{filename}: size: {f1_size} {f2_size} {change}')
     print(f'{filename}: common: {min_size} differs by {total_bytes_diff} bytes in {gran_blocks} granularity blocks')
     if gran_blocks != 0:
+        ts.files_changed += 1
         pct20_blocks = round(abs(pct20_count / gran_blocks * 100))
         pct80_blocks = round(abs(pct80_count / gran_blocks * 100))
         print(f'{filename}: smallest 20%: {pct20_count} of {gran_blocks} blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {granularity}')
         print(f'{filename}: largest 80%: {pct80_count} of {gran_blocks} blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {granularity}')
+    print("")
 
-def compare_backups(dir1, dir2, granularity):
+#
+# Print a detailed summary of all of the blocks and bytes over all of the files. We accumulated
+# the changes as we went through each of the files.
+#
+def print_summary():
+    print('SUMMARY')
+    # Calculate overall totals from each of the different types first.
+    total_blocks = 0
+    total_bytes = 0
+    total_files = 0
+    total_files_changed = 0
+    total_gran_blocks = 0
+    for t in global_types:
+        ts = typestats[t]
+        total_bytes += ts.bytes
+        total_files += ts.files
+        total_files_changed += ts.files_changed
+        total_gran_blocks += ts.gran_blocks
+    print(f'Total: {total_bytes} bytes changed in {total_gran_blocks} granularity-sized ({granularity}) blocks')
+    if total_files_changed == 1:
+        word = 'file'
+    else:
+        word = 'files'
+    print(f'Total: {total_files_changed} {word} changed out of {total_files} total files')
```

Comment:
Maybe cleaner with an extra function:
```
def plural(s, count):
    if count == 1:
       return s
    else
       return s + 's'

print(f'.....out of {total_files} total {plural("file", total_files_changed)}')
```
If that's too busy, you can use the function along with the `word` variable.

Reply:
I like something like that. I'll make some similar sort of change and see if it improves things. The repeated conditionals is not ideal.